client-app/shared/loyalty/composables/usePointsOperations.ts

```diff
@@ -0,0 +1,96 @@
+import { PointsOperationsSearchParams, registerPointsOperation, searchMyPointsOperations } from "@core/api/graphql/loyalty";
+import { Logger } from "@core/utilities";
+import { InputRegisterPointsOperationType, PointsOperationType, UserType } from "@core/api/graphql/types";
+import { MaybeRef } from "@vueuse/core";
+import { computed, readonly, ref, Ref, shallowRef, unref } from "vue";
+
+const fallbackItemsPerPage = 20;
+
+const loading: Ref<boolean> = ref(true);
+const loadingMore: Ref<boolean> = ref(false);
+const pointsOperations: Ref<PointsOperationType[]> = shallowRef([]);
+const total: Ref<number> = ref(0);
+const pages: Ref<number> = ref(1);
+
+export default (options: { userId: MaybeRef<string>, storeId: MaybeRef<string> }) => {
+  const { userId, storeId } = options;
+
+  async function fetchMyPointsOperations(searchParams: Partial<PointsOperationsSearchParams>) {
+    loading.value = true;
+    pointsOperations.value = [];
+    total.value = 0;
+    pages.value = 1;
+
+    searchParams.userId = unref(userId);
+    searchParams.storeId = unref(storeId);
+
+    try {
+      const {
+        items = [],
+        totalCount = 0,
+      } = await searchMyPointsOperations(searchParams);
+
+      pointsOperations.value = items;
+      total.value = totalCount;
+      pages.value = Math.ceil(totalCount / (searchParams.itemsPerPage || fallbackItemsPerPage));
+    } catch (e) {
+      Logger.error("usePointsOperations.fetchMyPointsOperations", e);
+      throw e;
+    } finally {
+      loading.value = false;
+    }
+  }
+
+  async function fetchMorePointsOperations(searchParams: Partial<PointsOperationsSearchParams>) {
+    loadingMore.value = true;
+
+    searchParams.userId = unref(userId);
+    searchParams.storeId = unref(storeId);
+
+    try {
+      const {
+        items = [],
+        totalCount = 0,
+      } = await searchMyPointsOperations(searchParams);
+
+      pointsOperations.value = pointsOperations.value.concat(items);
```

Comment:
Here you'll possible have a problem with key duplication in the template. There may appear some new records during scrolling, so some data may be duplicated. You'll have warnings in the console, cause `:key`s will be duplicated. Possible workarounds are:
1. Make a composite `:key` using the id and some ordinal number
2. Filter this array after the concat to remove duplicates.